core/store/src/trie/mod.rs

```diff
@@ -395,21 +396,21 @@ impl RawTrieNodeWithSize {
     }
 }
 
-fn encode_trie_node_with_rc(data: &[u8], rc: u32) -> Vec<u8> {
+fn encode_trie_node_with_rc(data: &[u8], rc: i32) -> Vec<u8> {
     let mut cursor = Cursor::new(Vec::with_capacity(data.len() + 4));
     cursor.write_all(data).unwrap();
-    cursor.write_u32::<LittleEndian>(rc).unwrap();
+    cursor.write_i32::<LittleEndian>(rc).unwrap();
     cursor.into_inner()
 }
 
-fn decode_trie_node_with_rc(bytes: &[u8]) -> Result<(&[u8], u32), StorageError> {
+fn decode_trie_node_with_rc(bytes: &[u8]) -> Result<(&[u8], i32), StorageError> {
```

Comment:
Why is it `i32` now?

Reply:
Update writes encoded positive or negative refcount delta, and merge adds up updates. Records/partial sum can be negative, but the result from db.get() should always be positive (if it is <= 0, there is a bug in gc). Probably need to add a new assert, this change deleted the one from apply_deletions.